### PR TITLE
generate and publish osg-cpu-hours metrics (SOFTWARE-4437)

### PR DESCRIPTION
**Pull request by edquist**
@bbockelm.

- pull in scripts used to generate osg-cpu-hours metrics
- add to nightly GHA run

### CODE REVIEW

**Review by brianhlin — COMMENTED** [review]
Two thoughts:

- May as well make this completely Python 3
- I know you're fond of the `bumbly.py` but I'd prefer a more descriptive like `gen-osg-cpu-table.py`

**Comment by edquist** [comment]
> May as well make this completely Python 3

I don't understand the request.  Is there something you would like changed?

File: osg-cpu-hours/bumbly.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python
```
Comment (brianhlin) [inline osg-cpu-hours/bumbly.py]:
```suggestion
#!/usr/bin/env python3
```

File: osg-cpu-hours/bumbly.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+from __future__ import print_function
+
```
Comment (brianhlin) [inline osg-cpu-hours/bumbly.py]:
```suggestion
```

File: osg-cpu-hours/cc_star_fqdns.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/python
```
Comment (brianhlin) [inline osg-cpu-hours/cc_star_fqdns.py]:
```suggestion
#!/usr/bin/python3
```

File: osg-cpu-hours/cc_star_fqdns.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+from __future__ import print_function
+
```
Comment (brianhlin) [inline osg-cpu-hours/cc_star_fqdns.py]:
```suggestion
```

**Review by brianhlin — COMMENTED** [review]
Some suggestions for what I meant to make it Python 3. I haven't looked closely enough to see if any other changes are required.

**Comment by edquist** [comment]
I guess i see no reason to mark them explicitly as python3 as long as they continue to work with python2.

As seen in the github workflow, all of the nightly metrics scripts will run under the same python(3) virtualenv.  That python will be python3, yes.  But if the same requirements are installed in a python2 virtualenv (eg outside of the github workflow environment), the `#!/usr/bin/env python` scripts will run fine there also.

If you have python3-specific features you want to use for some reason, fine - then i would agree with making the python3 explicit.  But for the moment i don't see a good reason to introduce python2-breaking changes.

**Comment by brianhlin** [comment]
> But for the moment i don't see a good reason to introduce python2-breaking changes.

Because I don't want to support Python 2 wherever possible.

**Comment by edquist** [comment]
>I know you're fond of the bumbly.py

Well you are right about that.  I don't know if i've ever given a program a more apt, descriptive name.

I'll see what i can do.

**Comment by edquist** [comment]
> I want us to go out of our way to introduce python2-breaking changes in order to make it clear to the world that we do not support python2

Very well then!  I will dutifully state my difference in opinion here.  Seems to me there is a difference between "not supporting" and "breaking compatibility".  I think it feels kinda lame to go out of our way to do the latter.

To adapt the words of Treebeard, "That doesn't make sense to me; but then, you are in charge."  :)

**Comment by brianhlin** [comment]
LGTM!

**Pull request merged**